docs/reference/high_resolution_pose_estimation.md

```diff
@@ -0,0 +1,385 @@
+  A list of integer scales that define the multiscale evaluation setup. Used to manually set the scales instead of going for the predefined multiscale setup.
+- **visualize**: *bool, default=False*\
+  Specifies whether the images along with the poses will be shown, one by one during evaluation.
+
+- **img_mean**: *list, default=(128, 128, 128)]*\
+  Specifies the mean based on which the images are normalized.
+- **img_scale**: *float, default=1/256*\
+  Specifies the scale based on which the images are normalized.
+- **pad_value**: *list, default=(0, 0, 0)*\
+  Specifies the pad value based on which the images' width is padded.
+- **half_precision**: *bool, default=False*\
+  Enables inference using half (fp16) precision instead of single (fp32) precision. Valid only for GPU-based inference.
+
+
+
+#### `HighResolutionPoseEstimationLearner.eval`
+```python
+HighResolutionPoseEstimationLearner.eval(self, dataset, silent, verbose, use_subset, subset_size, images_folder_name, annotations_filename)
+```
+
+This method is used to evaluate a trained model on an evaluation dataset.
+Returns a dictionary containing stats regarding evaluation.
+
+Parameters:
+
+- **dataset**: *object*\
+  Object that holds the evaluation dataset.
+  Can be of type `ExternalDataset` or a custom dataset inheriting from `DatasetIterator`.
+- **silent**: *bool, default=False*\
+  If set to True, disables all printing of evaluation progress reports and other information to STDOUT.
+- **verbose**: *bool, default=True*\
+  If set to True, enables the maximum verbosity.
+- **val_subset**: *bool, default=True*\
+  If set to True, a subset of the validation dataset is created and used in evaluation.
+- **subset_size**: *int, default=250*\
+  Controls the size of the validation subset.
+- **images_folder_name**: *str, default='val2017'*\
+  Folder name that contains the dataset images.
+  This folder should be contained in the dataset path provided.
+  Note that this is a folder name, not a path.
+- **annotations_filename**: *str, default='person_keypoints_val2017.json'*\
+  Filename of the annotations JSON file.
+  This file should be contained in the dataset path provided.
+
+#### `HighResolutionPoseEstimation.infer`
+```python
+HighResolutionPoseEstimation.infer(img, upsample_ratio, track, smooth)
+```
+
+This method is used to perform pose estimation on an image.
+Returns a list of `engine.target.Pose` objects, where each holds a pose, or returns an empty list if no detection were made.
+
+Parameters:
+
+- **img**: *object***\
+  Object of type engine.data.Image.
+- **upsample_ratio**: *int, default=4*\
+  Defines the amount of upsampling to be performed on the heatmaps and PAFs when resizing.
+- **track**: *bool, default=True*\
+  If True, infer propagates poses ids from previous frame results to track poses.
+- **smooth**: *bool, default=True*\
+  If True, smoothing is performed on pose keypoints between frames.
+
+
+#### `HighResolutionPoseEstimationLearner.first_pass`
```

Comment:
Probably need to add underscores to make them private and remove them from documentation. Nevertheless, i would suggest adding the documented explanation into docstrings on the methods, as it would be helpful for someone digging a bit deeper.